rust/geoparquet/src/writer/metadata.rs

```diff
@@ -171,13 +171,20 @@ impl GeoParquetMetadataBuilder {
     pub fn try_new(schema: &Schema, options: &GeoParquetWriterOptions) -> GeoArrowResult<Self> {
         let mut columns = HashMap::new();
 
+        let mut options_primary_column_seen = options.primary_column.is_none();
```

Comment:
IMO it would be simpler code and hardly less performant to iterate a second time to validate the primary column

```rs
if let Some(primary_column) = options.primary_column {
    if !schema
        .fields()
        .iter()
        .any(|field| field.name() == primary_column)
    {
        return Err(GeoArrowError::GeoParquet(format!(
            "Designated primary column: {} does not exist in schema",
            primary_column
        )));
    }
}
```

I'm not a big fan of the `if options_primary_column_seen` below because we have to remember how it relates to `options.primary_column` being `None`. Whereas using `if let Some()` ensures you have panic-free code.